packages/nimble-components/src/wafer-map/modules/computations.ts

```diff
@@ -101,7 +101,11 @@ export class Computations {
             this._containerDimensions.width,
             this._containerDimensions.height
         );
-        const gridDimensions = this.calculateGridDimensions(this.wafermap.dies);
+        // the dimensions are valid if they are all undefined
+        const gridDimensions = !this.wafermap.validity.invalidGridDimensions
+            && this.wafermap.gridMinX !== undefined
+            ? this.calculateGridDimensionsFromBoundingBox()
+            : this.calculateGridDimensionsFromDies(this.wafermap.dies);
```

Comment:
Instead of a comment and somewhat complex conditional you could make a well named function
 something like:

```ts
        const gridDimensions = this.gridDimensionsValidAndDefined()
            ? this.calculateGridDimensionsFromBoundingBox()
            : this.calculateGridDimensionsFromDies(this.wafermap.dies);
```